### ISSUE TITLE
Cookie size too large

### ISSUE THREAD

**Issue by wojodesign** [wojodesign/simplecart-js#15]
Need to create more cookies to handle more information when the cart has several items and the data is too large for one cookie

**Comment by lukevdp** [comment]
This is exactly the problem I'm having. Is anybody working on this?

**Comment by richtestani** [comment]
If you clear your paypal cookies, this fixes it but Im not sure what's corrupting the cookie.

**Comment by lukevdp** [comment]
Yep - I don't know enough to be able to fix this unfortunately. I did get around it in the short term by minimising the amount of info I was storing in the cookie, which allowed a heap more items to be added to the cart.

**Comment by lukevdp** [comment]
ok the code isn't very good, but it works untill the cookie size hits the LimitRequestFieldSize (in apache).

Replace the current "data storage and retrival" section with below.

```
/* load cart from cookie */
this.load = function () {
    /* initialize variables and items array */
    this.items = {};
    this.total = 0.00;
    this.quantity = 0;

    /* retrieve item data from cookie */
    if( readCookie('simpleCart1')){

        var data = unescape(readCookie('simpleCart1')).split('++');

        if(readCookie('simpleCart2')) {
            var data = unescape(readCookie('simpleCart1')+readCookie('simpleCart2')).split('++');
        }
        if (readCookie('simpleCart3'))
        {
            var data = unescape(readCookie('simpleCart1')+readCookie('simpleCart2')+readCookie('simpleCart3')).split('++');
        }
        if (readCookie('simpleCart4'))
        {
            var data = unescape(readCookie('simpleCart1')+readCookie('simpleCart2')+readCookie('simpleCart3')+readCookie('simpleCart4')).split('++');
        }


        for(var x=0, xlen=data.length;x<xlen;x++){

            var info = data[x].split('||');
            var newItem = new CartItem();

            if( newItem.parseValuesFromArray( info ) ){
                newItem.checkQuantityAndPrice();
                /* store the new item in the cart */
                this.items[newItem.id] = newItem;
            }
        }
    }
    this.isLoaded = true;
};



/* save cart to cookie */
this.save = function () {
    var dataString = "";
    var length_of_cookie_1 = 0;
    var length_of_cookie_2 = 0;
    var length_of_cookie_3 = 0;
    var count = 0;
    for( var item in this.items ){
        dataString = dataString + "++" + this.items[item].print();

        if (count <= 25) {
            var length_of_cookie_1 = length_of_cookie_1 + ("++" + this.items[item].print()).length;
        } else if (count > 25 && count <= 50) {
                var length_of_cookie_2 = length_of_cookie_2 + ("++" + this.items[item].print()).length;
        } else if (count >50 && count <= 75) {
            var length_of_cookie_3 = length_of_cookie_3 + ("++" + this.items[item].print()).length;
        }
        count = count + 1;
    }

    createCookie('simpleCart1', dataString.substring(2, length_of_cookie_1), 30);
    if (count > 25) {
        createCookie('simpleCart2', dataString.substring(length_of_cookie_1,length_of_cookie_2 + length_of_cookie_1), 30 );
    }
    if (count > 50) {
        createCookie('simpleCart3', dataString.substring(length_of_cookie_2 + length_of_cookie_1, length_of_cookie_3 + length_of_cookie_2 + length_of_cookie_1), 30 );
    }
    if (count > 75) {
        createCookie('simpleCart4', dataString.substring(length_of_cookie_3 + length_of_cookie_2 + length_of_cookie_1), 30 );
    }

};
```

**Comment by wojodesign** [comment]
Fixed in 2.2.dev
